app/src/main/java/org/jellyfin/androidtv/ui/playback/VideoManager.java

```diff
@@ -314,15 +321,47 @@ public long seekTo(long pos) {
         return pos;
     }
 
-    public void setVideoPath(@Nullable String path) {
+    private int getSubtitleSelectionFlags(MediaStream mediaStream) {
+        int flags = 0;
+        if (mediaStream.isDefault()) flags &= C.SELECTION_FLAG_DEFAULT;
+        if (mediaStream.isForced()) flags &= C.SELECTION_FLAG_FORCED;
+        return flags;
+    }
+
+    public void setMediaStreamInfo(ApiClient api, StreamInfo streamInfo) {
+        String path = streamInfo.getMediaUrl();
         if (path == null) {
             Timber.w("Video path is null cannot continue");
             return;
         }
         Timber.i("Video path set to: %s", path);
 
         try {
-            mExoPlayer.setMediaItem(MediaItem.fromUri(Uri.parse(path)));
+            // Add external subtitles
+            List<MediaItem.SubtitleConfiguration> subtitleConfigurations = new ArrayList<>();
+            for (MediaStream mediaStream : streamInfo.getMediaSource().getMediaStreams()) {
+                if (mediaStream.getType() != MediaStreamType.SUBTITLE) continue;
+
+                if (mediaStream.getDeliveryMethod() == SubtitleDeliveryMethod.EXTERNAL) {
+                    Uri subtitleUri = Uri.parse(api.createUrl(mediaStream.getDeliveryUrl(), Collections.emptyMap(), Collections.emptyMap(), true));
+                    MediaItem.SubtitleConfiguration subtitleConfiguration = new MediaItem.SubtitleConfiguration.Builder(subtitleUri)
+                            .setId("JF_EXTERNAL:" + String.valueOf(mediaStream.getIndex()))
+                            .setMimeType(SubtitleKt.getFfmpegSubtitleMimeType(mediaStream.getCodec()))
+                            .setLanguage(mediaStream.getLanguage())
+                            .setLabel(mediaStream.getDisplayTitle())
+                            .setSelectionFlags(getSubtitleSelectionFlags(mediaStream))
+                            .build();
+                    subtitleConfigurations.add(subtitleConfiguration);
+                    subtitleConfigurations.add(subtitleConfiguration);
```

Comment:
Did you mean to add it twice?

Reply:
Definitely not, that was for debugging. Good catch